test/unit/Cover/editCover.js

```diff
@@ -69,9 +70,75 @@ describe.skip('editCover', function () {
       coverAsset,
       period,
       amount: increasedAmount,
-      targetPriceRatio,
       gracePeriod,
-      segmentId: '1',
+      segmentId: 1,
+    });
+  });
+
+  it('should edit purchased cover and add coverage from a new staking pool', async function () {
+    const { cover } = this;
+
+    const [coverBuyer, manager] = this.accounts.members;
+
+    const { productId, coverAsset, period, amount } = coverBuyFixture;
+
+    const { expectedPremium, coverId: expectedCoverId } = await buyCoverOnOnePool.call(this, coverBuyFixture);
+
+    await createStakingPool(
+      cover,
+      productId,
+      parseEther('10000'), // capacity
+      coverBuyFixture.targetPriceRatio, // targetPrice
+      0, // activeCover
+      manager, // creator
+      manager, // manager
+      coverBuyFixture.targetPriceRatio, // currentPrice
+    );
+
+    const buyerBalanceBefore = await ethers.provider.getBalance(coverBuyer.address);
+
+    await cover.connect(coverBuyer).buyCover(
+      {
+        coverId: expectedCoverId,
+        owner: coverBuyer.address,
+        productId,
+        coverAsset,
+        amount,
+        period,
+        maxPremiumInAsset: expectedPremium.add(1),
+        paymentAsset: coverAsset,
+        payWitNXM: false,
+        commissionRatio: parseEther('0'),
+        commissionDestination: AddressZero,
+        ipfsData: '',
+      },
+      [
+        { poolId: 0, skip: true, coverAmountInAsset: 0 },
+        { poolId: 1, skip: false, coverAmountInAsset: amount },
```

Comment:
What is the expected behavior when all pool requests are set to `skip: true`? Does it make sense to add a test case for this scenario?

Reply:
Yup thats a really good idea. Will try to add that today.

Reply:
Added the test here https://github.com/NexusMutual/smart-contracts/pull/637